exporters/logging-otlp/src/main/java/io/opentelemetry/exporter/logging/otlp/internal/writer/StreamJsonWriter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.logging.otlp.internal.writer;
+
+import io.opentelemetry.exporter.internal.marshal.Marshaler;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.internal.ThrottlingLogger;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+ * any time.
+ */
+public class StreamJsonWriter implements JsonWriter {
+
+  private static final Logger internalLogger = Logger.getLogger(StreamJsonWriter.class.getName());
+
+  private final ThrottlingLogger logger = new ThrottlingLogger(internalLogger);
+
+  private final String type;
+  private final OutputStream originalStream;
+  private final OutputStream outputStream;
+
+  public StreamJsonWriter(OutputStream originalStream, String type) {
+    this.originalStream = originalStream;
+    this.outputStream = new IgnoreCloseOutputStream(originalStream);
```

Comment:
What's going on here? What close are we trying to ignore?

Also, what do you think in general about the resource management of the output stream? Should `OtlpStdoutLogRecordExporter` be responsible for calling `OutputStream.close()`? I think probably yes, since otherwise it would be complicated for a user to detect shutdown and close the output stream themselves. On the otherhand, closing `System.out` is not correct. I wonder if we need a boolean configuration option to indicate whether the `OutputStream` should be closed on `LogRecordExporter#shutdown()`:
```
setOutputStream(os, true);
```

Reply:
very good point - there's not even a test for using a file.

Reply:
turned out that we can skip the boolean flag if we just call `flush` in `close`

Reply:
Flushing on close will ensure that any unwritten bytes are flushed to the outputstream, but doesn't allow any outputstream resources to be released on shutdown.

Reply:
we could call close on shutdown instead - but maybe that's too surprising

another option is that we specifically check for `System.out` and `System.err` - and only then not call close (I like that).

Reply:
> we could call close on shutdown instead - but maybe that's too surprising

That was my point with adding a boolean on `setOutputStream` where the user could specify whether they want to be responsible for closing the stream or if they want the exporter to be responsible.

> another option is that we specifically check for System.out and System.err

True.. That would probably cover most of the cases. But I wonder if there are other cases where you would want to skip closing the `OutputStream`. We could always start with this suggestion and later add a boolean overload to `setOutputStream` if needed.

Reply:
> another option is that we specifically check for System.out and System.err

I implemented this now

Reply:
I see code in `IgnoreCloseOutputStream#close()` which checks if the stream is `System.out` or `System.err`, but I don't see where `close()` is being called anywhere in `OtlpStdoutLogRecordExporter#shutdown()`.

Also, we should add javadoc to `OtlpStdoutLogRecordExporterBuilder#setOutput(OutputStream)` which indicates that the stream will be closed if its not `System.out` or `System.err`.

Reply:
Great catch - I also found a better way to implement this I hope.

I couldn't find where `flush` is being called - and if I need to flush the stream manually.

Reply:
flush is called by jackson (https://fasterxml.github.io/jackson-core/javadoc/2.7/com/fasterxml/jackson/core/JsonGenerator.Feature.html#FLUSH_PASSED_TO_STREAM) - but it seems that https://github.com/open-telemetry/opentelemetry-java/blob/8a917e036c823b17d13b8a535d104218318df931/sdk/logs/src/main/java/io/opentelemetry/sdk/logs/export/LogRecordExporter.java#L72 is unused